KubeArmor/feeder/policyMatcher.go

```diff
@@ -12,440 +13,338 @@ import (
+	match := tp.MatchPolicy{
+		PolicyName: policyName,
+		Severity:   strconv.Itoa(netProt.Severity),
+		Tags:       netProt.Tags,
+		Message:    netProt.Message,
+		Source:     source,
+		Operation:  "Network",
+		Resource:   resource,
+		Action:     netProt.Action,
+	}
 
-						match.Action = dir.Action
+	return match
+}
 
-						matches.Policies = append(matches.Policies, match)
-					} else {
-						for _, src := range dir.FromSource {
-							if len(src.Path) > 0 {
-								match := tp.MatchPolicy{}
+func getProtocolFromName(proto string) string {
+	switch strings.ToLower(proto) {
+	case "tcp":
+		return "type=SOCK_STREAM"
+	case "udp":
+		return "type=SOCK_DGRAM"
+	case "icmp":
+		return "type=SOCK_RAW protocol=1"
+	default:
+		return ""
+	}
+}
 
-								match.PolicyName = secPolicy.Metadata["policyName"]
+func newCapMatchPolicy(policyName, source, resource string, cap tp.CapabilitiesCapabilityType) tp.MatchPolicy {
+	match := tp.MatchPolicy{
+		PolicyName: policyName,
+		Severity:   strconv.Itoa(cap.Severity),
+		Tags:       cap.Tags,
+		Message:    cap.Message,
+		Source:     source,
+		Operation:  "Network",
+		Resource:   resource,
+		Action:     cap.Action,
+	}
 
-								match.Severity = strconv.Itoa(dir.Severity)
-								match.Tags = dir.Tags
-								match.Message = dir.Message
+	return match
+}
 
-								match.Source = src.Path
-								match.Operation = "Process"
-								match.Resource = dir.Directory
+func getCapabilityFromName(cap string) string {
+	switch strings.ToLower(cap) {
+	case "net_raw":
+		return "type=SOCK_RAW protocol=1"
+	default:
+		return ""
+	}
+}
 
-								match.Action = dir.Action
+// UpdateSecurityPolicies Function
+func (fd *Feeder) UpdateSecurityPolicies(action string, conGroup tp.ContainerGroup) {
+	name := conGroup.NamespaceName + "_" + conGroup.ContainerGroupName
 
-								matches.Policies = append(matches.Policies, match)
-							} else if len(src.Directory) > 0 {
-								match := tp.MatchPolicy{}
+	if action == "DELETED" {
+		delete(fd.SecurityPolicies, name)
+		return
+	}
 
-								match.PolicyName = secPolicy.Metadata["policyName"]
+	// ADDED | MODIFIED
+	matches := tp.MatchPolicies{}
 
-								match.Severity = strconv.Itoa(dir.Severity)
-								match.Tags = dir.Tags
-								match.Message = dir.Message
+	for _, secPolicy := range conGroup.SecurityPolicies {
+		policyName := secPolicy.Metadata["policyName"]
 
-								match.Source = src.Directory
-								match.Operation = "Process"
-								match.Resource = dir.Directory
+		for _, path := range secPolicy.Spec.Process.MatchPaths {
+			fromSource := ""
 
-								match.Action = dir.Action
+			if len(path.FromSource) == 0 {
+				match := newProcPathMatchPolicy(policyName, fromSource, path)
+				matches.Policies = append(matches.Policies, match)
+				continue
+			}
 
-								matches.Policies = append(matches.Policies, match)
-							}
-						}
-					}
+			for _, src := range path.FromSource {
```

Comment:
This is a XOR checking as I've mentioned on the main PR [discussion](https://github.com/accuknox/KubeArmor/pull/152#issuecomment-862447832).